### PR TITLE
box model crud

### PR DESCRIPTION
**Pull request by svileshina**
Co-authored-by: Robb Kidd <robb@thekidds.org>
Co-authored-by: Mae Beale <maebeale@gmail.com>

creating box model crud

### CODE REVIEW

**Review by robbkidd — APPROVED** [review]
tenor-94968427

**Review by robbkidd — CHANGES_REQUESTED** [review]
tenor-38617954


Oops. We need to alias the referenced models in the `Box` model class, too.

File: app/models/box.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Box < ApplicationRecord
+  belongs_to :box_request
+  belongs_to :designed_by
+  belongs_to :design_reviewed_by
+  belongs_to :assembled_by
+  belongs_to :shipped_by
+  belongs_to :shipping_payment
```
Comment (h-m-m) [inline app/models/box.rb]: I pulled this locally and could not `#create!` a box with a nil in any of :designed_by :design_reviewed_by, :assembled_by, :shipped_by, or :shipping_payment. These should all be nullable

Comment (robbkidd) [reply app/models/box.rb]: Right!

File: app/models/purchase.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Purchase < ApplicationRecord
+  belongs_to :location
```
Comment (gabezurita) [inline app/models/purchase.rb:2]: @maebeale If you would like to add receipt_file_upload capabilities to this model, just include the following:
```suggestion
  has_many_attached :receipt_file_uploads
  belongs_to :location
```

I think a Purchase can have many receipts? That's not clear to me. If it's just one, then you can do:
`has_one_attached :receipt_file_upload`

File: app/assets/javascripts/boxes.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// Place all the behaviors and hooks related to the matching controller here.
```
Comment (gabezurita) [inline app/assets/javascripts/boxes.js]: @maebeale, everything in the /assets folder in this PR can be deleted:
```
boxes.js
purchases.js
boxes.css
purchases.css
```

**Review by robbkidd — APPROVED** [review]
tenor-244381438

**Review by gabezurita — APPROVED** [review]
Looks great!

File: app/models/purchase.rb (hunk 2 above)
Comment (gabezurita) [reply app/models/purchase.rb:2]: will take care of this in:
https://github.com/rubyforgood/voices-of-consent/issues/75

**Pull request merged**